cmd/system-probe/main_common.go

```diff
@@ -87,20 +92,47 @@ func runAgent() {
 		cleanupAndExit(1)
 	}
 
-	sysprobe, err := CreateSystemProbe(cfg)
-	if err != nil && strings.HasPrefix(err.Error(), ErrSysprobeUnsupported.Error()) {
+	conn, err := net.NewListener(cfg)
+	if err != nil {
+		log.Criticalf("Error creating IPC socket: %s", err)
+		cleanupAndExit(1)
+	}
+
+	loader := NewLoader()
+	httpMux := http.NewServeMux()
+
+	err = loader.Register(cfg, httpMux, factories)
+	if err != nil && strings.HasPrefix(err.Error(), modules.ErrSysprobeUnsupported.Error()) {
 		// If tracer is unsupported by this operating system, then exit gracefully
 		log.Infof("%s, exiting.", err)
 		gracefulExit()
-	} else if err != nil {
+	}
+	if err != nil {
```

Comment:
Curious, but why get rid of the `else if`?

Reply:
just style 🤷‍♂️ 
I I think that if a conditional branch ends in a `return` or in this case a `gracefulExit()`, it's more readable if you get rid of the else block.
In fact that is what the official golang linter suggests: https://github.com/golang/lint/blob/738671d3881b9731cc63024d5d88cf28db875626/lint.go#L980-L981